osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -625,6 +625,15 @@ private void delete(BeatmapSetInfo beatmap)
             dialogOverlay?.Push(new BeatmapDeleteDialog(beatmap));
         }
 
+        private void clearScores(BeatmapSetInfo beatmap)
+        {
+            if (beatmap == null || beatmap.ID <= 0) return;
```

Comment:
you aren't checking if the user is looking at local scores here. there's a high chance you're incorrectly passing online scores from the leaderboard to be deleted.

Reply:
Added this in https://github.com/ppy/osu/pull/4000/commits/472325b885b4ba54e2b9c4a7383c0d2997298d51